File: 10. Subprogram/E. Komposisi Fungsi/e_komposisi_fungsi.cpp
#include <iostream>

int fx(int A, int B, int x){
  return abs(A * x + B);
}

int main(){
  int A, B, K, x;
  std::cin >> A >> B >> K >> x;
  while (K--){
    x = fx(A, B, x);
  }
  std::cout << x << std::endl;
  return 0;
}

File: 10. Subprogram/E. Komposisi Fungsi/e_komposisi_fungsi.rs
fn fx(a: i32, b: i32, x: i32) -> i32 {
    let res = a * x + b;
    return res.abs();
}

fn main() {
    let mut reader = String::new();
    std::io::stdin().read_line(&mut reader).expect("");
    let mut substr_iter = reader.split_whitespace();
    let mut next_num = || -> i32 { substr_iter.next().expect("").parse().expect("") };
    let a = next_num();
    let b = next_num();
    let k: i32 = next_num();
    let mut x: i32 = next_num();
    for _ in 0..k {
        x = fx(a, b, x);
    }
    println!("{}", x);
}
